applications/FSIapplication/python_scripts/fsi_analysis.py

```diff
@@ -0,0 +1,338 @@
+
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.MeshMovingApplication as MeshMovingApplication
+import KratosMultiphysics.FluidDynamicsApplication as FluidDynamicsApplication
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+try:
+    import KratosMultiphysics.ExternalSolversApplication
+except ImportError:
+    pass
+
+from analysis_stage import AnalysisStage
+
+class FSIAnalysis(AnalysisStage):
+    '''Main script for FSI simulations using the FSI family of python solvers.'''
+
+    def __init__(self, model, project_parameters):
+        '''The constructor of the FSI analysis object
+
+        Note that the base class analysis stage constructor
+        is not called intentionally, since it is not compatible
+        with the current FSI application Json structure.
+        '''
+        if (type(model) != Kratos.Model):
+            raise Exception("Input is expected to be provided as a Kratos Model object")
+
+        if (type(project_parameters) != Kratos.Parameters):
+            raise Exception("Input is expected to be provided as a Kratos Parameters object")
+
+        self.model = model
+        self.project_parameters = project_parameters
+
+        echo_fluid = self.project_parameters["fluid_solver_settings"]["problem_data"]["echo_level"].GetInt()
+        echo_structure = self.project_parameters["structure_solver_settings"]["problem_data"]["echo_level"].GetInt()
+        self.echo_level = max(echo_fluid, echo_structure)
+        self.parallel_type = self.project_parameters["coupling_solver_settings"]["problem_data"]["parallel_type"].GetString()
+
+        # If this is an MPI run, load the distributed memory modules
+        if (self.parallel_type == "MPI"):
+            from KratosMultiphysics.mpi import mpi
+            import KratosMultiphysics.MetisApplication
+            import KratosMultiphysics.TrilinosApplication
+            import KratosMultiphysics.MappingApplication #TODO: Import always once we use the serial version of the mapper
+            self.is_printing_rank = (mpi.rank == 0)
+        else:
+            self.is_printing_rank = True
+
+        # Deprecation warnings
+        # This makes possible the FSI solver derivation from the core base python_solver.py
+        if not self.project_parameters.Has("echo_level"):
+            Kratos.Logger.PrintInfo("FSIAnalysis", "Using the old way to pass the echo_level, this will be removed!")
+            self.project_parameters.AddEmptyValue("echo_level")
+            self.project_parameters["echo_level"].SetInt(self.echo_level)
+
+        # Add solver variables (note that the solver is created in the first _GetSolver() call)
+        self._GetSolver().AddVariables()
+
+    def Initialize(self):
+        '''
+        Construct and initialize all classes and tools used in the simulation loop.
+        '''
+        self._SetUpRestart()
+
+        if self.load_restart:
```

Comment:
The restart is now handled differently => such that not every stage has to copy paste the code from the other ones

I sent @RiccardoRossi a working example yesterday, it is much simpler

Then I think you could use `Initilaize` from the baseclass

Reply:
I will have a look. Thanks @philbucher